### PR TITLE
Add helper methods for complete-on-view to Service

### PR DESCRIPTION
**Pull request by symbolist**
**Description:** Moving some helpers method from https://github.com/edx/edx-platform/pull/18273 to the Completion Service.

**JIRA:** Link to JIRA ticket

**Dependencies:** None

**Installation instructions:** N/A

**Testing instructions:** N/A

**Reviewers:**
- [x] @jcdyer 
- [x] @iloveagent57 

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [x] Commits are squashed
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: completion/tests/test_services.py [hunk 1]
```diff
@@ -80,6 +84,67 @@ def test_enabled_honors_waffle_switch(self, enabled):
...
+        block.has_score = has_score
+        if has_custom_completion:
+            block.has_custom_completion = True
+
+        self.assertEqual(self.completion_service.can_mark_block_complete_on_view(block), can_mark_complete)
+
+    def test_blocks_to_mark_complete_on_view(self):
+
+        block_1 = XBlock(Mock(), scope_ids=Mock(spec=ScopeIds))
+        block_1.location = UsageKey.from_string("i4x://edX/100/a/1").replace(course_key=self.course_key)
```
Comment (jcdyer) [inline completion/tests/test_services.py]: Is this use of replace the same as `usage_key.map_into_course(course_key)`?

File: completion/tests/test_services.py [hunk 2]
```diff
@@ -80,6 +84,67 @@ def test_enabled_honors_waffle_switch(self, enabled):
...
+        block_2 = XBlock(Mock(), scope_ids=Mock(spec=ScopeIds))
+        block_2.location = UsageKey.from_string("i4x://edX/100/a/2").replace(course_key=self.course_key)
+        block_3 = XBlock(Mock(), scope_ids=Mock(spec=ScopeIds))
+        block_3.location = UsageKey.from_string("i4x://edX/100/a/3").replace(course_key=self.course_key)
+        block_3.completion_mode = XBlockCompletionMode.AGGREGATOR
```
Comment (jcdyer) [inline completion/tests/test_services.py]: Nit: Consider giving these blocks descriptive names (E.g., s/block_3/aggregator_block/).

File: completion/tests/test_services.py [hunk 3]
```diff
@@ -89,7 +154,7 @@ class CompletionDelayTestCase(CompletionSetUpMixin, TestCase):
     """
 
     @ddt.data(1, 1000, 0)
-    def test_get_completion_by_viewing_delay_ms(self, delay):
+    def test_get_complete_on_view_delay_ms(self, delay):
```
Comment (jcdyer) [inline completion/tests/test_services.py]: Have you made sure you caught all uses of this method in edx-platform?

File: completion/tests/test_services.py (hunk 1 above)
Comment (symbolist) [reply completion/tests/test_services.py]: I suspect so. I just copied this pattern from line 37 above.

File: completion/tests/test_services.py (hunk 3 above)
Comment (symbolist) [reply completion/tests/test_services.py]: Yup. They are used in two views and there is is bokchoy coverage for both.

File: completion/tests/test_services.py (hunk 2 above)
Comment (symbolist) [reply completion/tests/test_services.py]: Updated names.

**Comment by symbolist** [comment]
@iloveagent57 Can you please merge this as well? And if I create a tag, will that push a release to PyPi?

**Comment by iloveagent57** [comment]
@symbolist yup, merging now.  And yes, if you create a tag (I usually just use the github release feature) it will kick off a travis build that will push a new release to PyPI.

**Comment by symbolist** [comment]
@iloveagent57 Thanks! Looks like I do not have access to either push a tag or create a release. Can you please take care of this?

**Comment by iloveagent57** [comment]
@symbolist yup, will do (I'll also make a note of this for future PRs).

**Comment by iloveagent57** [comment]
https://pypi.org/project/edx-completion/ is now at the latest version, 0.1.7

**Comment by symbolist** [comment]
@iloveagent57 Thanks!

**Comment by edx-webhook** [comment]
Although this pull request is already merged, I've created [OSPR-5543](https://openedx.atlassian.net/browse/OSPR-5543) so that we can track it in Jira.

There is nothing you have to do. No action is needed from your side.  Thanks again for your contribution.

**Pull request merged**